src/mesh/index_derivs.cxx

```diff
@@ -1881,7 +1881,7 @@ const Field3D Mesh::indexVDDX(const Field3D &v, const Field3D &f, CELL_LOC outlo
     Mesh::flux_func func = sfVDDX;
     DiffLookup *table = UpwindTable;
 
-    if (vloc == CELL_XLOW) {
+    if ((vloc == CELL_XLOW) && (inloc == CELL_CENTRE)) {
```

Comment:
I think this change means that the following snippet:

```cpp
    if ((vloc == CELL_XLOW) && (inloc == CELL_CENTRE)) {
      // V staggered w.r.t. variable
      func = sfVDDX;
      table = UpwindStagTable;
      diffloc = CELL_CENTRE;
    } else if ((vloc == CELL_CENTRE) && (inloc == CELL_XLOW)) {
      // Shifted
      func = sfVDDX;
      table = UpwindStagTable;
      diffloc = CELL_XLOW;
    } else {
 ```

could be rewritten as:

```cpp
    if (((vloc == CELL_XLOW) && (inloc == CELL_CENTRE))
        || ((vloc == CELL_CENTRE) && (inloc == CELL_XLOW))){
      // V staggered w.r.t. variable
      func = sfVDDX;
      table = UpwindStagTable;
    } else {
```

As now `diffloc` is explicitly the same as `inloc`, which is set at 1868. Unless I am missing something?

The same applies to the `if` statements at 2272, 2538, etc.

Reply:
There were quite a few of these. I've cut them down in 380d891, as well as getting rid of all the 'diffloc's.